File: tests/fixtures/fake_jaxrs_app/src/main/java/com/example/api/ItemResource.java
package com.example.api;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

@Path("/api/items")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class ItemResource {

    @GET
    public List<Item> listItems(@QueryParam("category") String category) {
        return List.of();
    }

    @GET
    @Path("/{id}")
    public Item getItem(@PathParam("id") Long id) {
        return new Item();
    }

    @POST
    public Item createItem(@QueryParam("name") String name) {
        return new Item();
    }

    @PUT
    @Path("/{id}")
    public Item updateItem(@PathParam("id") Long id, @QueryParam("name") String name) {
        return new Item();
    }

    @DELETE
    @Path("/{id}")
    public void deleteItem(@PathParam("id") Long id) {
    }
}

File: tests/fixtures/fake_kotlin_jaxrs_app/src/main/kotlin/com/example/api/ItemResource.kt
package com.example.api

import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("/api/items")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
class ItemResource {

    @GET
    fun listItems(@QueryParam("category") category: String?): List<Item> {
        return emptyList()
    }

    @GET
    @Path("/{id}")
    fun getItem(@PathParam("id") id: Long): Item {
        return Item()
    }

    @POST
    fun createItem(@QueryParam("name") name: String): Item {
        return Item()
    }

    @PUT
    @Path("/{id}")
    fun updateItem(@PathParam("id") id: Long, @QueryParam("name") name: String?): Item {
        return Item()
    }

    @DELETE
    @Path("/{id}")
    fun deleteItem(@PathParam("id") id: Long) {
    }
}
